singularity-eos/eos/modifiers/ramps_eos.hpp

```diff
@@ -171,6 +186,16 @@ class SAPRampEOS : public EosBase<SAPRampEOS<T>> {
     }
     // call internal filleos
     t_.FillEos(rho, temp, energy, press, cv, bmod, ramp_out, lambda);
+    // fill ramp density
+    Real rho_ramp{rho};
+    if (output & thermalqs::density) {
+      assert(!(output & thermalqs::pressure));
+      rho_ramp = get_ramp_density(press);
+    }
+    rho = rho_ramp > rho ? rho_ramp : rho;
```

Comment:
Starting a new code comment to keep things clean.

I think the logic here is backwards. The ramp is used to decrease the EOS density so you only want to use it when it's less than the EOS density.